pkg/handler/metricsconfigurer/handler_test.go

```diff
@@ -0,0 +1,390 @@
+			expectedResult: nil,
+			expectedError:  true,
+		},
+		{
+			name:           "Empty response",
+			responseBody:   []byte{},
+			expectedResult: nil,
+			expectedError:  true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			result, err := h.parseResponse(tc.responseBody, mockRequest)
+
+			if tc.expectedError {
+				if err == nil {
+					t.Errorf("Expected error but got nil for testcase %s", tc.name)
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Unexpected error: %v for testcase %s", err, tc.name)
+				}
+
+				// Check the result
+				if !reflect.DeepEqual(result, tc.expectedResult) {
+					t.Errorf("Unexpected result for testcase %s. Got: %+v, Expected: %+v", tc.name, result, tc.expectedResult)
+				}
+			}
+		})
+	}
+}
+
+// test getDatasource
+func TestGetDatasource(t *testing.T) {
+	// Create a handler with mocked values
+	h := Handler{
+		Logger:        logr.Discard(),
+		AccountNumber: "123456789",
+		DatasourceID:  "testDatasourceID",
+	}
+
+	// Create a mock HTTP server
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer mockServer.Close()
+
+	// Create a mock request
+	req := &Request{
+		RequestType:       "Create",
+		ResponseURL:       mockServer.URL,
+		StackId:           "arn:aws:cloudformation:us-east-1:123456789:stack/MyStack/guid",
+		RequestId:         "unique_id",
+		ResourceType:      "Custom::MyCustomResource",
+		LogicalResourceId: "MyCustomResource",
+	}
+
+	// Prepare test data
+	token := "testToken"
+	domainName := "observe-eng.com"
+
+	// Create a mock client
+	mockRoundTripper := &mockRoundTripper{
+		RoundTripFunc: func(req *http.Request) (*http.Response, error) {
+			// Check the request
+			if diff := cmp.Diff("POST", req.Method); diff != "" {
+				t.Fatalf("incorrect method: %s", diff)
+			}
+			if diff := cmp.Diff("application/json", req.Header.Get("Content-Type")); diff != "" {
+				t.Fatalf("incorrect content-type: %s", diff)
+			}
+			if diff := cmp.Diff("https://123456789.observe-eng.com/v1/meta", req.URL.String()); diff != "" {
+				t.Fatalf("incorrect url: %s", diff)
+			}
+			if diff := cmp.Diff("Bearer 123456789 testToken", req.Header.Get("Authorization")); diff != "" {
+				t.Fatalf("incorrect authorization: %s", diff)
+			}
+
+			// Prepare mock response
+			mockResponse := &http.Response{
+				StatusCode: 200,
+				Body: io.NopCloser(bytes.NewBufferString(`{
+                    "data": {
+                        "datasource": {
+                            "name": "TestDatasource",
+                            "variables": [
+                                {
+                                    "name": "Metrics",
+                                    "details": {
+                                        "awsCollectionStackConfig": {
+                                            "awsServiceMetricsList": [
+                                                {
```

Comment:
might be good to include multiple metrics here to make sure we're mapping over them all